src/picongpu/include/fields/currentInterpolation/Binomial/Binomial.hpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2015 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "types.h"
+#include "basicOperations.hpp"
+
+#include "fields/currentInterpolation/None/None.def"
+
+namespace picongpu
+{
+namespace currentInterpolation
+{
+using namespace PMacc;
+
+template<uint32_t T_dim>
+struct Binomial
+{
+    static const uint32_t dim = T_dim;
+
+    typedef typename PMacc::math::CT::make_Int<dim, 1>::type LowerMargin;
+    typedef typename PMacc::math::CT::make_Int<dim, 1>::type UpperMargin;
+
+    template<typename DataBoxE, typename DataBoxB, typename DataBoxJ>
+    HDINLINE void operator()(DataBoxE fieldE,
+                             DataBoxB,
+                             DataBoxJ fieldJ )
+    {
+        const DataSpace<dim> self;
+        typedef typename DataBoxJ::ValueType TypeJ;
+
+        /* 1 2 1 weighting for "left"(1x) "center"(2x) "right"(1x),
+         * see Pascal's triangle level N=2 */
+        TypeJ dirSum( TypeJ::create(0.0) );
+        for( uint32_t d = 0; d < dim; ++d )
+        {
+            DataSpace<dim> dw;
```

Comment:
If you need const `DataSpace<>` this is the way. (only a not, no need to put it in)
```C++   
 typedef typename PMacc::math::CT::make_Vector<
        simDim,
        bmpl::integral_c<int,0>
    >::type ZeroVector;

    typedef typename PMacc::math::CT::Assign<
        ZeroVector,
        bmpl::integral_c<int,0>,
        bmpl::integral_c<int,-1>
    >::type Minus;

    typedef typename PMacc::math::CT::Assign<
        ZeroVector,
        bmpl::integral_c<int,0>,
        bmpl::integral_c<int,1>
    >::type Plus;

        const PMACC_AUTO(minus,Minus::toRT());
        const PMACC_AUTO(plus,Plus::toRT());
```